src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,632 @@
+               - Any CE specific string with additional parameters for MP jobs
+        "ARCRESTTimeout"    - DEfault = 1.0 (seconds)
+               - Timeout for the rest query
+        "proxyTimeLeftBeforeRenewal" - Default = 10000 (seconds)
+               - As the name says
+
+        Note : This is not run from __init__ as the design of DIRAC means that ceParameters is
+        filled with CEDefaults only at the time this class is initialised for the given CE
+        """
+        # super()._reset()
+        self.log.debug("Testing if the REST interface is available", "for %s" % self.ceName)
+
+        # Get options from the ceParameters dictionary
+        self.port = self.ceParameters.get("Port", self.port)
+        self.restVersion = self.ceParameters.get("RESTVersion", self.restVersion)
+        self.queue = self.ceParameters.get("Queue", self.queue)
+
+        self.proxyTimeLeftBeforeRenewal = self.ceParameters.get(
+            "proxyTimeLeftBeforeRenewal", self.proxyTimeLeftBeforeRenewal
+        )
+        self.arcRESTTimeout = self.ceParameters.get("ARCRESTTimeout", self.arcRESTTimeout)
+
+        # Build the URL based on the CEName, the port and the REST version
+        service_url = "https://" + self.ceName + ":" + self.port
+        self.base_url = service_url + "/arex/rest/" + self.restVersion + "/"
+
+        # Set up the request framework
+        self.session = requests.Session()
+        self.session.verify = Locations.getCAsLocation()
+        self.headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        # Attach the token to the headers if present
+        if os.environ.get("BEARER_TOKEN"):
+            self.headers["Authorization"] = "Bearer " + os.environ["BEARER_TOKEN"]
+
+        return S_OK()
+
+    #############################################################################
+
+    def _arcToDiracID(self, arcJobID):
+        """Convert an ARC jobID into a DIRAC jobID.
+        Example: 1234 becomes https://<ce>:<port>/arex/1234
+
+        :param str: ARC jobID
+        :return: DIRAC jobID
+        """
+        # Add CE and protocol information to pilot ID
+        if "://" in arcJobID:
+            self.log.warn("Identifier already in ARC format", arcJobID)
+            return arcJobID
+
+        diracJobID = "https://" + self.ceHost + ":" + self.port + "/arex/" + arcJobID
+        return diracJobID
+
+    def _DiracToArcID(self, diracJobID):
+        """Convert a DIRAC jobID into an ARC jobID.
+        Example: https://<ce>:<port>/arex/1234 becomes 1234
+
+        :param str: DIRAC jobID
+        :return: ARC jobID
+        """
+        # Remove CE and protocol information from pilot ID
+        if "://" in diracJobID:
+            arcJobID = diracJobID.split("arex/")[-1]
+            return arcJobID
+        self.log.warn("Identifier already in REST format?", diracJobID)
+        return diracJobID
+
+    def _UrlJoin(self, words):
```

Comment:
Indeed, we control the content. However, it has been a repeated request and so has been implemented this way.

I did take a close look at urlparse. However, it can only join exactly two strings. There is also urlunsplit which does something very different with exactly 6 arguments. And os.path.join does something very different.

So, after googling a bit, I found that writing a function gave me the best solution. However, if you can implement a similar feature with a standard library, I am happy to use it!